### ISSUE TITLE
syntaxnet: Unicode error on non-ASCII input

### ISSUE THREAD

**Issue by Zert** [tensorflow/models#318]
## syntaxnet
Hello

I've built an docker image for syntaxnet. All I have after startup with non-ASCII sentences is unicode error:
```
$ docker run  syntaxnet sh -c 'echo "Повесть была принята издателем." | syntaxnet/demo.sh'
...
Traceback (most recent call last):
  File "/opt/tensorflow/models/syntaxnet/bazel-bin/syntaxnet/conll2tree.runfiles/syntaxnet/conll2tree.py", line 93, in <module>
    tf.app.run()
  File "/opt/tensorflow/models/syntaxnet/bazel-bin/syntaxnet/conll2tree.runfiles/external/org_tensorflow/tensorflow/python/platform/app.py", line 30, in run
    sys.exit(main(sys.argv))
  File "/opt/tensorflow/models/syntaxnet/bazel-bin/syntaxnet/conll2tree.runfiles/syntaxnet/conll2tree.py", line 84, in main
    print 'Input: %s' % sentence.text
UnicodeEncodeError: 'ascii' codec can't encode characters in position 7-13: ordinal not in range(128)
```

ASCII-only input works pretty well.

**Comment by girving** [comment]
@calberti Should syntaxnet work correctly for unicode?

**Comment by zeryx** [comment]
I can confirm we have the exact same problem, painful problem to address without replacing the conll2tree.py file.

**Comment by zhichaohu** [comment]
I'm having a similar problem. But for me it seems that SyntaxNet can produce correct dependency parse for unicode text: it successfully parsed an English corpus with characters like "Café" in it (commented out the last 4 lines in demo.sh for dependency parse). However it threw out this "can't encode" error when I was trying to do the regular parse. 

GitHub pages says that "The script (demo.sh) will break the text into words, run the POS tagger, run the parser, and then **generate an ASCII version of the parse tree**." So I'm guessing it failed at converting unicode characters to ascii when generating the parse tree?

**Comment by andorardo** [comment]
Sorry you came across such a strange error! Unicode should work fine, even with `asciitree`. The Cyrillic example works without error when Parsey is run from the command line. The error seems to be related to piping through Docker.

Allocating a pseudo-TTY with the `-t` flag:
 ```
docker run -t syntaxnet sh -c 'echo "Повесть была принята издателем." | syntaxnet/demo.sh'
```
should fix the error.

**Comment by zeryx** [comment]
Hey @andorardo, that solution only works with simple implementations of the docker image, our temporary solution was to fork tensorflow/models and add `.encode('UTF-8')`  to the two `print` lines in the conll2tree.py file.
https://github.com/zeryx/models/

This pull request also seems to fix it:
https://github.com/tensorflow/models/pull/292

**Comment by calberti** [comment]
Merged #292, thanks zeryx!
